apps/back-office/src/app/dashboard/pages/users/users.component.ts

```diff
@@ -24,31 +34,133 @@ export class UsersComponent implements OnInit {
   public users = new Array<User>();
   public roles: Role[] = [];
   public displayedColumns = ['name', 'username', 'oid', 'roles', 'actions'];
+  private usersQuery!: QueryRef<GetUsersQueryResponse>;
+  public cachedUsers: User[] = [];
+
+  public pageInfo = {
+    pageIndex: 0,
+    pageSize: ITEMS_PER_PAGE,
+    length: 0,
+    endCursor: '',
+  };
 
   /**
    * UsersComponent constructor.
    *
    * @param apollo Used to load the users.
+   * @param translate Translate service
    */
-  constructor(private apollo: Apollo) {}
+  constructor(private apollo: Apollo, private translate: TranslateService) {}
 
   /** Load the users */
   ngOnInit(): void {
-    this.apollo
-      .watchQuery<GetUsersQueryResponse>({
-        query: GET_USERS,
-      })
-      .valueChanges.subscribe((resUsers) => {
-        this.loading = true;
-        this.users = resUsers.data.users;
-        this.apollo
-          .watchQuery<GetRolesQueryResponse>({
-            query: GET_ROLES,
+    this.usersQuery = this.apollo.watchQuery<GetUsersQueryResponse>({
+      query: GET_USERS,
+      variables: {
+        first: ITEMS_PER_PAGE,
+        afterCursor: null,
+      },
+    });
+    this.usersQuery.valueChanges.subscribe((resUsers) => {
```

Comment:
Hi @RenzoPrats , now that we are adding new query subscriptions, please add the unsubscribe teardown logic for those(SafeUnsubscribeComponent),

```suggestion
    this.usersQuery.valueChanges.pipe(takeUntil(this.destroy$).subscribe((resUsers) => {
```

Same for the query of roles.
Thanks